source/developer-guides/debug.rst

```diff
@@ -0,0 +1,197 @@
+.. _debug:
+
+Source Level Debugging
+--------------
+
+This explains how to perform basic debugging operation with Intel UDK Debugger on SBL over serial port. (used Apollo Lake (ABL) as example)
+
+
+Prerequisite
+^^^^^^^^^^^^^^
+
+* Install |Intel UDK Debugger|, using the Intel |copy| UDK Debugger Tool on a Windows platform requires a host machine configured with: 
+
+
+    * Microsoft Win7 x64 or Win8 x64  or Win10 x64
+    
+    * Microsoft Windows Debug Tool(WinDbg) for proper Windows version.     
+  
+
+.. |Intel UDK Debugger| raw:: html
+
+   <a href="https://firmware.intel.com/develop/intel-uefi-tools-and-utilities/intel-uefi-development-kit-debugger-tool#overlay-context=develop" target="_blank">Intel UDK Debugger on HOST</a>
+
+  
+* The Intel |copy| UDK Debugger can be downloaded |here| for Windows and Linux, install Intel UDK Debugger with default configuration values.  
+
+.. |here| raw:: html
+
+   <a href="https://firmware.intel.com/develop/intel-uefi-tools-and-utilities/intel-uefi-development-kit-debugger-tool#overlay-context=develop" target="_blank">here</a>  
+  
+* Install |Windows Debug Tool(WinDbg)| with default configuration values.  
+
+.. |Windows Debug Tool(WinDbg)| raw:: html
+
+   <a href="https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/debugger-download-tools" target="_blank">Windows Debug Tool(WinDbg)</a>  
+
+General Steps
+^^^^^^^^^^^^^^
+
+**Step 1 - Enable Debug Agent**
+
+Enable DebugAgent on SBL and flash the SBL image(or IFWI) on TARGET. By default, DebugAgent is not enabled. It can be simply enabled with a single line change.
```

Comment:
For SBL and APL references, use corresponding name in substitutions.txt

Reply:
fixed